packages/grafana-data/src/field/fieldColor.test.ts

```diff
@@ -27,4 +44,47 @@ describe('fieldColorModeRegistry', () => {
     const calcFn = getCalculator({ mode: FieldColorModeId.PaletteClassic, seriesIndex: 1 });
     expect(calcFn(70, 0, undefined)).toEqual('#EAB839');
   });
+
+  it('When color.seriesBy is set to last use that instead of v', () => {
+    const field = getTestField('continuous-GrYlRd');
+
+    field.config.color!.seriesBy = 'last';
+    // min = -10, max -10, last: 5
+    // last percent 75%
+    field.values = new ArrayVector([0, -10, 5, 10, 2, 5]);
+
+    const color = getFieldSeriesColor(field, getTestTheme());
+    const calcFn = getCalculator({ mode: 'continuous-GrYlRd' });
+
+    expect(color.color).toEqual(calcFn(4, 0.75));
```

Comment:
Took me some time to get this in my head right and I'm not 100 sure I succeeded:) Can you leave a comment in the code where does 4 come from?

Reply:
can't really explain this one, the last value is 5, not a 4, confused : )

Reply:
@dprokop ok now I understand it, the 4 can be anything as this color scale only uses the percent value 0.75 comes from the fact that 5 is 75% in the range -10 to 10

Reply:
added comment to this line

Reply:
heh, is see the confusion hit not only me:) thanks a lot!